src/bit-systems/waypoint.ts

```diff
@@ -164,6 +170,20 @@ export function waypointSystem(
       }
     }
 
+    // When a scene is opened with a named waypoint we have to make sure that the scene default waypoint
+    // doesn't override it and that we correctly spawn in the named waypoint from the url.
+    // https://github.com/mozilla/hubs/issues/2833
+    const hashUpdated = window.location.hash !== "" && previousWaypointHash !== window.location.hash;
+    if (hashUpdated && initialSpawnHappened) {
+      const waypointName = window.location.hash.replace("#", "");
+      const waypointObj = world.eid2obj.get(eid);
+      if (waypointObj && waypointObj.name === waypointName) {
```

Comment:
Nit: The existence of `waypointObj` is not checked below at https://github.com/mozilla/hubs/blob/c60c6cbb9b5452c81d1c28fcc86efede9ccca126/src/bit-systems/waypoint.ts#L203-L204 so no need to check here for the consistency, or should be checked for the both.

Reply:
Actually that code was there before and we haven't had crashes so I assume we expect that object to exist. I'm going to remove the guard.